src/org/montclairrobotics/sprocket/control/ArcadeDriveInput.java

```diff
@@ -1,53 +1,60 @@
 package org.montclairrobotics.sprocket.control;
 
-import edu.wpi.first.wpilibj.Joystick;
+import org.montclairrobotics.sprocket.drive.DriveTrainTarget;
 import org.montclairrobotics.sprocket.drive.MotorInputType;
-import org.montclairrobotics.sprocket.drive.DriveTrainInput;
+import org.montclairrobotics.sprocket.geometry.Angle;
 import org.montclairrobotics.sprocket.geometry.Degrees;
 import org.montclairrobotics.sprocket.geometry.Inch;
 import org.montclairrobotics.sprocket.geometry.Polar;
 import org.montclairrobotics.sprocket.geometry.Vector;
+import org.montclairrobotics.sprocket.geometry.XY;
+import org.montclairrobotics.sprocket.pipeline.Step;
+
+import edu.wpi.first.wpilibj.Joystick;
 
-public class ArcadeDriveInput extends DriveTrainInput {
+public class ArcadeDriveInput implements Step<DriveTrainTarget> {
```

Comment:
Then maybe in the Pipeline class could we just have a "setInput" method which sets the first step in the pipeline and replaces whatever was previously in the first spot?